### PR TITLE
[MNT] test in with linux up to python 3.11

### LINKED ISSUE #22: properly set up matrix os for CI
Tackle after merging of #18

> the other errors came either from changing v2->v3 or from matrix os (I guess the latter).

yes most likely the latter because the way it is set up now, all tests are only run on Linux

See this workflow that is supposed to run on MacOS but actually ran on Ubuntu
https://github.com/DeepMReye/DeepMReye/runs/7576038296?check_suite_focus=true#step:1:3

So I would suggest merging this and fixing the CI for OS in another PR after that.

_Originally posted by @Remi-Gau in https://github.com/DeepMReye/DeepMReye/issues/18#issuecomment-1200162849_

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes #22

### CODE REVIEW

**Comment by Remi-Gau** [comment]
good news: tests seem to pass on linux python 3.11

**Comment by Remi-Gau** [comment]
@Naubody @CYHSM 
I would suggest to merge this and open an issue to remember to check if CI can be set up for MacOS in the future.

Possibly a good thing to cut a new release to let people use a version that works with python 3.11.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
-        os: [ubuntu-latest, macOS-10.15, windows-latest]
-        python-version: ["3.8", "3.9"] 
-        include:
-          - os: ubuntu-latest
-            path: ~/.cache/pip
-          - os: macOS-10.15
-            path: ~/Library/Caches/pip
-          - os: windows-latest
-            path: ~\AppData\Local\pip\Cache
+        os: [ubuntu-latest] # MacOS fails to run at all # windows-latest: does not work because antspyx is not available for windows
```
Comment (CYHSM) [inline .github/workflows/main.yml:19]: Previously the CI ran on Windows but could be that it only runs in the CI environment (have not tested it on a Windows machine). MacOS should work and we have also tested it, so not sure whats the error you get? Related to 3.11?

Comment (Remi-Gau) [reply .github/workflows/main.yml:19]: Actually the CI has always run on only ubuntu latest. This what the workflow looked like at the last release and before the merge of this PR.

https://github.com/DeepMReye/DeepMReye/blob/a8ae38c1fda09dbdf5bcb3aef083a94e818dfd97/.github/workflows/main.yml#L12

The issue was mentioned there:
https://github.com/DeepMReye/DeepMReye/issues/22

Eventhough there was a matrix set up for set up for all OS, the `run-on` was always set on ubuntu.

If you try to run on all OSes or on on Mac-OS latest like was done here:

- https://github.com/DeepMReye/DeepMReye/commit/2fe89991def9925d60c11067197b1e33fc61072c
- https://github.com/DeepMReye/DeepMReye/commit/a06b653777d1afec421f32b0c5673ae6ac03dd41


You get these errors:

- for windows:

https://github.com/DeepMReye/DeepMReye/actions/runs/5061393380/jobs/9085545204#step:4:1466

```
   ERROR: Failed building wheel for antspyx
  Building wheel for deepmreye (pyproject.toml): started
  Building wheel for deepmreye (pyproject.toml): finished with status 'done'
  Created wheel for deepmreye: filename=deepmreye-0.1.1-py3-none-any.whl size=2870948 sha256=6aebaaaecb177162ebfb0889d183941aa726751a42ad8754de2fae398f064c5a
  Stored in directory: C:\Users\runneradmin\AppData\Local\Temp\pip-ephem-wheel-cache-63flyek8\wheels\50\5c\e3\cd5a5e6a6ffa429bbbd1173e5c88079df5e39b0da06d5585a7
  Building wheel for jax (pyproject.toml): started
  Building wheel for jax (pyproject.toml): finished with status 'done'
  Created wheel for jax: filename=jax-0.4.10-py3-none-any.whl size=1480617 sha256=c7d5cab68c4fb3f54a5216336ac455b9a149014d152cd28a65a3f2d653200e5e
  Stored in directory: c:\users\runneradmin\appdata\local\pip\cache\wheels\af\ad\a6\2732752b23e24d5de10c9aabe0d73e31220fc2492af1fa19bd
  Building wheel for wrapt (pyproject.toml): started
  Building wheel for wrapt (pyproject.toml): finished with status 'done'
  Created wheel for wrapt: filename=wrapt-1.14.1-cp311-cp311-win_amd64.whl size=35494 sha256=3ecbc06c2a037aa18ec38042563100fcf5ee82e12b840a8802dc081dd18dc874
  Stored in directory: c:\users\runneradmin\appdata\local\pip\cache\wheels\eb\b6\fa\5ab6f4107cad63fa04c54ad78d75bb7035119bdd4f751df5ae
Successfully built deepmreye jax wrapt
Failed to build antspyx
ERROR: Could not build wheels for antspyx, which is required to install pyproject.toml-based projects
```

- MacOS

-  the workflow ran for 6 hours and was killed by Github. 

https://github.com/DeepMReye/DeepMReye/actions/runs/5061833053/jobs/9086577565

I can rerun it to recheck or set up a PR with a matrix with MacOS that we can re-run to see if it goes anywhere at some point.

**Pull request merged**